index.html

```diff
@@ -296,19 +301,19 @@ <h2>
           then Y. It is RECOMMENDED that axes appear in decreasing order of
           importance, such that element 0 and 1 typically represent the X and Y
           axis of a directional stick. The same object MUST be returned until
-          the <a>user agent</a> needs to return different values (or values in
-          a different order).
+          the <a>user agent</a> needs to modify the length of the array or
+          return different axis values.
         </dd>
         <dt>
           <dfn>buttons</dfn> attribute
         </dt>
         <dd>
-          Array of button states for all buttons of the gamepad. It is
+          Array of GamepadButton objects representing all buttons of the gamepad. It is
           RECOMMENDED that buttons appear in decreasing importance such that
           the primary button, secondary button, tertiary button, and so on
           appear as elements 0, 1, 2, ... in the buttons array. The same object
-          MUST be returned until the <a>user agent</a> needs to return
-          different values (or values in a different order).
+          MUST be returned until the <a>user agent</a> needs to modify the
+          length of the array or the button ordering.
```

Comment:
This is somewhat ambiguous... it could be read like, if an axis changes, then replace all every `GamepadButton`.

Reply:
I've rewritten the description of the axes and buttons attributes to make it clearer when the array should be reused and when it should be recreated.